### PR TITLE
BaSP-TG-100: Form - Employee

### PR DESCRIPTION
**Pull request by matescarabino**
- Add a redux thunk to GET the entity by its ID
- Change some logic to make form work with new getByIdEmployee() thunk
- Implement react-hook-form
- Implement Joi validations and resolver

**How to test QA**:
1- Navigate to employee entity in the navbar

2- Test **create** employee work as intended
    2.1- Test new "reset" button, it should reset every field to blank
    2.2- Test error messages (check ticket TG-100 Joi validations)

3- Test **update** employee work as intended
    3.1- Test new "reset" button, it should reset every field to the original value
    3.2- Test error messages (check ticket TG-100 Joi validations)

### CODE REVIEW

**Review by CristianBucci — APPROVED** [review]
Everything looks right to me!

**Review by Ansitox — APPROVED** [review]
Looks good to me

File: src/Components/Employees/Form/validations.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import Joi from 'joi';
+
+export const employeeSchema = Joi.object({
+  name: Joi.string()
+    .required()
+    .pattern(/^[\p{L}]+$/u)
+    .min(3)
+    .messages({
+      'string.pattern.base': 'All characters must be letters.'
```
Comment (ignacioaristo) [inline src/Components/Employees/Form/validations.js:9]: Custom all the messages (min and requierd too)

File: src/Components/Employees/Form/validations.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }),
+  lastName: Joi.string()
+    .required()
+    .pattern(/^[\p{L}]+$/u)
+    .min(3)
+    .messages({
+      'string.pattern.base': 'All characters must be letters.'
```
Comment (ignacioaristo) [inline src/Components/Employees/Form/validations.js:16]: same in all the validations

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/Components/Employees/Form/validations.js (hunk 1 above)
Comment (matescarabino) [reply src/Components/Employees/Form/validations.js:9]: done

File: src/Components/Employees/Form/validations.js (hunk 2 above)
Comment (matescarabino) [reply src/Components/Employees/Form/validations.js:16]: done

**Review by ignacioaristo — APPROVED** [review]

**Review by julianpiaggio — COMMENTED** [review]
🕵️‍♂️ QA FEEDBACK

We wanted to tell you that we consider that your pr complies
the requirements to be approved, and that it can be merged into the master branch 👍

**Pull request merged**